python/cucim/src/cucim/skimage/_shared/distance.py

```diff
@@ -0,0 +1,177 @@
+import math
+import warnings
+
+import cupy as cp
+import numpy as np
+
+_distance_on_cpu = True
+try:
+    from cupyx.scipy.spatial import distance
+
+    try:
+        test_points = cp.zeros((10, 2), dtype=cp.int32)
+        distance.cdist(test_points, test_points)
+        del test_points
+        _distance_on_cpu = False
+    except (RuntimeError, AttributeError, TypeError, ValueError) as e:
+        print(f"CuPy distance computation test failed with error: {e}")
+except ImportError as e:
+    print(f"cupyx.scipy.spatial.distance import failed with error: {e}")
+    pass
+
+if _distance_on_cpu:
+    import scipy.spatial.distance as distance
+
+
+__all__ = ["pdist_max_blockwise"]
+
+
+def pdist_max_blockwise(
+    coords,
+    metric="sqeuclidean",
+    *,
+    coords_per_block=4000,
+    compute_argmax=False,
+    cdist_kwargs={},
+):
+    """Find maximum pointwise distance.
+
+    Computes by processing blocks of coordinates to reduce overall memory
+    requirement. The memory used at runtime will be proportional to
+    ``coords_per_block**2``.
+
+    A block size of >= 2000 is recommended to overhead poor GPU resource usage
+    and to reduce kernel launch overhead.
+
+    Parameters
+    ----------
+    coords : np.ndarray (num_points, ndim)
+        The coordinates to process.
```

Comment:
The code converts input coordinates to float32 by default, which could lead to precision loss. I think it would be good to note that this `coords` would be converted to `float32` data type.

Reply:
I added a note that distances are always computed in 32-bit precision. Currently this utility is only used internally for the "feret_distance_max" region property. Given that the object is already discretized to pixels, the distance measurements based on pixel coordinates will already be a pretty rough measurement that is only accurate to ~1 pixel spacing, so high precision should not be needed.